calendar.js

```diff
@@ -26,32 +26,60 @@ function getScheduleAndUpdateTable(table)
         cache: false,
         success: function (data) {
             $.each(data, function (i, item) {
-                time[i] = item.time.length;
-            });
+                time[i] = item.time;
+             });
             updateTable(table, time);
         }
     });
 }
-
+//function to update the table according to the booked appointments
 function updateTable(table,time)
 {
-    for (var i = 0; i < 24; i++) {
-
+    for (var i = 0; i < 24; i++)
+    {
```

Comment:
break this function down into two parts:
1) function to create the initial empty calendar with the hour of day displayed
2) function that will update the empty calendar with the schedule.

combining these two concerns will make it harder to debug them separately. Also, other methods (say create alert in future) will need to operate directly on the table. Having a lot of table-creation logic entangled with the appointment updating logic makes it difficult to modify